apps/api/e2e/api/environment/guard-check.e2e.ts

```diff
@@ -0,0 +1,44 @@
+import { UserSession } from '@novu/testing';
+import { expect } from 'chai';
+import { NotificationTemplateRepository } from '@novu/dal';
+import { CreateNotificationTemplateDto } from '../../../src/app/notification-template/dto/create-notification-template.dto';
+
+describe('Environment - Check Root Environment Guard', async () => {
+  let session: UserSession;
+  const notificationTemplateRepository: NotificationTemplateRepository = new NotificationTemplateRepository();
+
+  before(async () => {
+    session = new UserSession();
+    await session.initialize();
+  });
+
+  it('should not allow create when not in development environment', async () => {
+    const testTemplate: Partial<CreateNotificationTemplateDto> = {
+      name: 'test template',
+      description: 'This is a test description',
+      notificationGroupId: session.notificationGroups[0]._id,
+      steps: [],
+    };
+
+    const { body: envsBody } = await session.testAgent.get('/v1/environments');
+
+    const envs = envsBody.data;
+
+    for (const env of envs) {
```

Comment:
A small suggestion would be not to test with a for loop, since if the envs will return empty array this test will still pass. I would suggestion doing a .find on the envs array and assert on it. @ainouzgali